Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -1827,13 +1825,46 @@ public void setIdentity(@NonNull String userId, @Nullable BranchReferralInitList
         }
     }
 
+    /**
+     * Gets all available cross platform ids.
+     *
+     * @param callback An instance of {@link io.branch.referral.util.BranchCrossPlatformId.BranchCrossPlatformIdListener}
+     *                to callback with cross platform ids
+     *
+     */
     public void getCrossPlatformIds(BranchCrossPlatformIdListener callback) {
         new BranchCrossPlatformId(callback, context_);
     }
 
+    /**
+     * Gets the available last attributed touch data. Defaults to a 30 day attribution window, and sends -1 to request the default from the server.
+     *
+     * @param callback An instance of {@link io.branch.referral.ServerRequestGetLATD.BranchLastAttributedTouchDataListener}
+     *                 to callback with last attributed touch data
+     *
+     */
+    public void getLastAttributedTouchData(BranchLastAttributedTouchDataListener callback) {
```

Comment:
I would like to see the following unit test pass
```
    @Test
    public void testGetLATDVariation() throws Throwable {
        Branch branch = Branch.getInstance(getTestContext());
        PrefHelper prefHelper = PrefHelper.getInstance(getTestContext());

        // Get Variation A
        branch.getLastAttributedTouchData(null, 100);
        int prefValueA = prefHelper.getLATDAttributionWindow();

        Assert.assertEquals(100, prefValueA);

        // Get Variation B
        branch.getLastAttributedTouchData(null);
        int prefValueB = prefHelper.getLATDAttributionWindow();

        Assert.assertEquals(prefValueA, prefValueB);
    }
```

Reply:
This test assumes that `branch.getLastAttributedTouchData(null, 100);` would set the attribution window preference, which is an undesired effect, so I wrote two new tests instead but they test the same scenario as here: calling `getLATDAttributionWindow` without custom set attribution window does not change the pref value.